src/decaylanguage/decay/decay.py

```diff
@@ -6,11 +6,21 @@
 
 from collections import Counter
 from copy import deepcopy
+from typing import (
+    Any,
+    Iterable,
+    Iterator,
+    Optional,
+    Type,
+    TypeVar,
+    Union,
+    no_type_check,
+)
 
 from ..utils import charge_conjugate_name
 
 
-class DaughtersDict(Counter):
+class DaughtersDict(Counter[str]):
```

Comment:
This is likely not subclass able until 3.8 or 3.9 - in fact, the ability to customize `__mro_entries__` to allow non-classes here, and the ability to have a `__class_getitem__`, wasn't added until 3.7. I'd do:

```python
if typing.TYPE_CHECKING:
    CounterStr = Counter[str]
else:
    CounterStr = Counter
```

Or similar.

Reply:
OK, will commit this in the next batch.